inc/computer.class.php

```diff
@@ -117,6 +133,105 @@ function defineTabs($options=array()) {
    }
 
 
+   /**
+    * @param $item         CommonGLPI object
+    * @param $tabnum       (default 1)
+    * @param $withtemplate (default 0)
+   **/
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum=1, $withtemplate=0) {
+
+      self::showOperatingSystem($item);
+      return true;
+   }
+
+   /**
+    * Print the computer's operating system form
+    *
+    * @param $comp Computer object
+    *
+    * @return Nothing (call to classes members)
+   **/
+   static function showOperatingSystem(Computer $comp) {
+      global $DB;
+
+      $ID = $comp->fields['id'];
+      if (!$comp->can($ID, UPDATE)) {
+        return false;
+      }
+
+
+      $colspan = 4;
+
+      echo "<div class='center'>";
+
+      $comp->initForm($ID);
+      $comp->showFormHeader();
+
+      if ($comp->isNewID($ID)) {
```

Comment:
This will never happen (tabs not displayed for new objects). And if needed, should be NOT (new ID have no value ;)